local/network.go

```diff
@@ -560,23 +560,14 @@ func (ln *localNetwork) Stop(ctx context.Context) error {
 
 // Assumes [ln.lock] is held.
 func (ln *localNetwork) stop(ctx context.Context) error {
-	ctx, cancel := context.WithTimeout(ctx, stopTimeout)
-	defer cancel()
 	errs := wrappers.Errs{}
 	for nodeName := range ln.nodes {
-		select {
-		case <-ctx.Done():
-			// In practice we'll probably never time out here,
-			// and the caller probably won't cancel a call
-			// to stop(), but we include this to respect the
-			// network.Network interface.
-			return ctx.Err()
-		default:
-		}
-		if err := ln.removeNode(ctx, nodeName); err != nil {
+		stopCtx, stopCtxCancel := context.WithTimeout(ctx, stopTimeout)
+		if err := ln.removeNode(stopCtx, nodeName); err != nil {
```

Comment:
Previously, all calls to `removeNode` shared a context and timeout, so if one node timed out, all other nodes would be SIGKILLed. Now, each node gets its own timeout so this won't happen.